### PR TITLE
[doc] Troubleshooting --dashboard-port

### PR DESCRIPTION
**Pull request by dHannasch**
See https://github.com/ray-project/ray/pull/11808 first.

## Why are these changes needed?

Currently, https://docs.ray.io/en/master/configure.html#head-node says that `--dashboard-port` is required to be open, but doesn't say exactly when it is required to be open or what problem you will see if it *isn't* open.
## Related issue number

It seemed more fruitful to open with the suggested change rather than complain about what wasn't there, but there is a related *pull request*: because https://github.com/ray-project/ray/pull/11808 was going to have merge conflicts with this, this actually incorporates https://github.com/ray-project/ray/pull/11808 so that this won't get tripped up by merge conflicts if https://github.com/ray-project/ray/pull/11808 is merged.

So there's actually an unrelated addition of `--redis-shard-ports` sitting in this branch, which is why https://github.com/ray-project/ray/pull/11808 should be looked at first. (It's much shorter and simpler than this.)
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/configure.rst [hunk 1]
```diff
@@ -135,9 +135,31 @@ Head Node
 In addition to ports specified above, the head node needs to open several more ports.
 
 - ``--port``: Port of GCS. Default: 6379.
-- ``--dashboard-port``: Port for accessing the dashboard. Default: 8265
+- ``--redis-shard-ports``: Comma-separated list of ports for non-primary Redis shards. Default: Random values.
 - ``--gcs-server-port``: GCS Server port. GCS server is a stateless service that is in charge of communicating with the GCS. Default: Random value.
 
+- If ``--include-dashboard`` is true (the default), then the head node must open ``--dashboard-port``. Default: 8265.
+
+If ``--include-dashboard`` is true but the ``--dashboard-port`` is not open on
```
Comment (rkooo567) [inline doc/source/configure.rst:143]: Thanks for the PR!! It is really helpful :).

Btw, this is probably something we should resolve (although it makes sense to document now as it is not working now). @mfitton Do you have any idea why this happens?

**Comment by richardliaw** [comment]
merged the other PR :)

**Review by richardliaw — APPROVED** [review]
Thanks a bunch, @dHannasch !

File: doc/source/configure.rst (hunk 1 above)
Comment (mfitton) [reply doc/source/configure.rst:143]: I don't know why it happens. I'll create an item to look into it next sprint.

**Pull request merged**